clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -642,17 +642,19 @@ public void sendOffsetsToTransaction(Map<TopicPartition, OffsetAndMetadata> offs
      * This method should be used when you need to batch consumed and produced messages
      * together, typically in a consume-transform-produce pattern. Thus, the specified
      * {@code groupMetadata} should be extracted from the used {@link KafkaConsumer consumer} via
-     * {@link KafkaConsumer#groupMetadata()} to leverage consumer group metadata for proper fencing.
+     * {@link KafkaConsumer#groupMetadata()} to leverage consumer group metadata for stronger fencing than
+     * {@link #sendOffsetsToTransaction(Map, String)} which only sends with consumer group id.
```

Comment:
`which only sends with consumer group id` -- seems obvious from the API. Should we remove this part?

Reply:
I feel keeping it is fine, just a more explicit explanation to the `stronger fencing`